### PR TITLE
PS-7015: Remove lz4 submodule and use bundled lz4 (8.0)

### PR DESCRIPTION
**Pull request by inikep**
Remove `lz4` submodule for RocksDB and use bundled lz4 1.9.2 instead.

### CODE REVIEW

File: storage/rocksdb/CMakeLists.txt [hunk 1]
```diff
@@ -119,18 +119,14 @@ STRING(REGEX MATCHALL "[^\n]+" ROCKSDB_LIB_SOURCES ${SCRIPT_OUTPUT})
 # add bundled compression code
 SET(ROCKSDB_LIB_SOURCES
   ${ROCKSDB_LIB_SOURCES}
-  ${CMAKE_CURRENT_SOURCE_DIR}/third_party/lz4/lib/lz4.c
-  ${CMAKE_CURRENT_SOURCE_DIR}/third_party/lz4/lib/lz4hc.c
-  ${CMAKE_CURRENT_SOURCE_DIR}/third_party/lz4/lib/lz4frame.c
-  ${CMAKE_CURRENT_SOURCE_DIR}/third_party/lz4/lib/xxhash.c
+  ${CMAKE_SOURCE_DIR}/extra/lz4/xxhash.c
```
Comment (percona-ysorokin) [inline storage/rocksdb/CMakeLists.txt:122]: Why is this different than in 5.7?

Comment (inikep) [reply storage/rocksdb/CMakeLists.txt:122]: In 5.7 we are using XXHash from `zstd` submodule with:
```
${CMAKE_CURRENT_SOURCE_DIR}/third_party/zstd/lib/common/xxhash.c
```
In 8.0 we removed `zstd` submodule.

The second option for 8.0 is using `extra/zstd/lib/common/xxhash.c` (XXHash 0.6.2) but `extra/lz4/xxhash.c` is newer with XXHash 0.6.5.

**Review by percona-ysorokin — APPROVED** [review]
LGTM

**Comment by inikep** [comment]
Tested at https://ps80.cd.percona.com/job/percona-server-8.0-param/650/

**Pull request merged**